src/nodes/lighting/AnalyticLightNode.js

```diff
@@ -99,24 +98,19 @@ class AnalyticLightNode extends LightingNode {
 
 	}
 
-	/**
-	 * Overwrites the default {@link Node#customCacheKey} implementation by including the
-	 * `light.id` and `light.castShadow` into the cache key.
-	 *
-	 * @return {number} The custom cache key.
-	 */
-	customCacheKey() {
```

Comment:
@sunag I have removed this code since it wasn't executed once in my tests. Under what circumstances does this code run? AFAICS, the code that detects light state changes must go into `LightsNode.customCacheKey()`.

Reply:
When looking at the code it may be out of date, since the material lights are being calculated entirely in LightsNode.